src/dft/backends/backend_wrappers.cxx

```diff
@@ -0,0 +1,92 @@
+        oneapi::mkl::dft::BACKEND::compute_forward<                                          \
+            oneapi::mkl::dft::detail::descriptor<PRECISION, DOMAIN>, T_REAL,                 \
+            T_REAL>, /* Buffer API */                                                        \
+        oneapi::mkl::dft::BACKEND::compute_backward<                                         \
+            oneapi::mkl::dft::detail::descriptor<PRECISION, DOMAIN>, T_BACKWARD>,            \
+        oneapi::mkl::dft::BACKEND::compute_backward<                                         \
+            oneapi::mkl::dft::detail::descriptor<PRECISION, DOMAIN>, T_REAL>,                \
+        oneapi::mkl::dft::BACKEND::compute_backward<                                         \
+            oneapi::mkl::dft::detail::descriptor<PRECISION, DOMAIN>, T_BACKWARD, T_FORWARD>, \
+        oneapi::mkl::dft::BACKEND::compute_backward<                                         \
+            oneapi::mkl::dft::detail::descriptor<PRECISION, DOMAIN>, T_REAL,                 \
+            T_REAL>, /* USM API */                                                           \
+        oneapi::mkl::dft::BACKEND::compute_backward<                                         \
+            oneapi::mkl::dft::detail::descriptor<PRECISION, DOMAIN>, T_BACKWARD>,            \
+        oneapi::mkl::dft::BACKEND::compute_backward<                                         \
+            oneapi::mkl::dft::detail::descriptor<PRECISION, DOMAIN>, T_REAL>,                \
+        oneapi::mkl::dft::BACKEND::compute_backward<                                         \
+            oneapi::mkl::dft::detail::descriptor<PRECISION, DOMAIN>, T_BACKWARD, T_FORWARD>, \
+        oneapi::mkl::dft::BACKEND::compute_backward<                                         \
+            oneapi::mkl::dft::detail::descriptor<PRECISION, DOMAIN>, T_REAL, T_REAL>,
+
+    ONEAPI_MKL_DFT_BACKEND_SIGNATURES(oneapi::mkl::dft::detail::precision::SINGLE,
+                                      oneapi::mkl::dft::detail::domain::REAL, float, float,
+                                      std::complex<float>)
+        ONEAPI_MKL_DFT_BACKEND_SIGNATURES(oneapi::mkl::dft::detail::precision::SINGLE,
+                                          oneapi::mkl::dft::detail::domain::COMPLEX, float,
+                                          std::complex<float>, std::complex<float>)
+            ONEAPI_MKL_DFT_BACKEND_SIGNATURES(oneapi::mkl::dft::detail::precision::DOUBLE,
+                                              oneapi::mkl::dft::detail::domain::REAL, double,
+                                              double, std::complex<double>)
+                ONEAPI_MKL_DFT_BACKEND_SIGNATURES(oneapi::mkl::dft::detail::precision::DOUBLE,
+                                                  oneapi::mkl::dft::detail::domain::COMPLEX, double,
+                                                  std::complex<double>, std::complex<double>)
```

Comment:
The indentation looks off, maybe because of the trailing `,` at the end of the macro. I think we could disable clang-format for this block.